DOCSP 13860: UpdateMany usage example (#7)
mongodb/docs-golang: source/usage-examples/updateMany.txt

=========================

.. default-domain:: mongodb

You can update multiple documents in a collection by using the ``UpdateMany()``
method.

The following example passes a query filter and an update parameter to the
``UpdateMany()`` method, which matches documents in the
``listingsAndReviews`` collection where the value of the ``address.market``
field is "Sydney" and multiplies the ``price`` field in the matching
documents by a factor of *1.15*:

.. include:: /includes/run-example-tip.rst

.. literalinclude:: /code-snippets/updateMany.go
   :start-after: begin updatemany
   :end-before: end updatemany
   :emphasize-lines: 5
   :language: go
   :dedent:

> Click here **<TODO: github link to file>** to see a fully runnable example.

Reviewer: I assume similar to https://github.com/mongodb/docs-golang/pull/5#discussion_r676970318 this link, and the TODO link below will be updated before publishing.

=========================

.. default-domain:: mongodb

You can update multiple documents in a collection by using the ``UpdateMany()``
method.

The following example passes a query filter and an update parameter to the

> ``UpdateMany()`` method. The query filter matches documents in the
> ``listingsAndReviews`` collection if the value of their "address.market"

Reviewer 2: ```suggestion
``UpdateMany()`` method, which matches documents in the
``listingsAndReviews`` collection with the value of their ``address.market``
```
Suggestion: Keeping the wording consistent with your UpdateOne page, and field names should be monospaced.

Author: accepted, but I think the word "if" should stay in place of "with". Does the sentence seem too long now?

Reviewer 2: A bit. In the past, we converted sentences that seemed long to either a few shorter sentences or a bulleted list of steps/action.

field is "Sydney", and the update filter multiplies the ``price`` field
in the matching documents by a factor of *1.15*:

.. include:: /includes/run-example-tip.rst

.. literalinclude:: /code-snippets/updateMany.go
   :start-after: begin updatemany
   :end-before: end updatemany
   :emphasize-lines: 5
   :language: go
   :dedent:

Click here **<TODO: github link to file>** to see a fully runnable example.

Expected Result
---------------

After you run the code example, the ``price`` field of the matched
documents will reflect the multiplication. The updated documents will
appear similar to this one:

.. code-block:: json
   :copyable: false

   // result truncated
   { 
      "_id" : "17438054",
      ...
      "name":"Cozy Granny Flat",
      ...
      "price": "115.00"
      ...

>    }

Reviewer 2: Question: Since this is an updateMany operation, can we show at least 2 documents updated (the first and last)? A potential solution is to make each updated document one line so it won't take too much vertical space.

Author: added two more examples and formatted as one line. LMK if you have suggestions on formatting the ellipses.

=========================

.. default-domain:: mongodb

You can update multiple documents in a collection by using the ``UpdateMany()``
method.

The following example passes a query filter and an update parameter to the
``UpdateMany()`` method, which matches documents in the
``listingsAndReviews`` collection if the value of their ``address.market``
field is "Sydney" and multiplies the ``price`` field in the matching
documents by a factor of *1.15*:

.. include:: /includes/run-example-tip.rst

.. literalinclude:: /code-snippets/updateMany.go
   :start-after: begin updatemany
   :end-before: end updatemany
   :emphasize-lines: 5
   :language: go
   :dedent:

Click here **<TODO: github link to file>** to see a fully runnable example.

Expected Result
---------------

After you run the code example, the ``price`` field of the matched
documents will reflect the multiplication. The updated documents will
appear similar to these:

.. code-block:: json
   :copyable: false

   // results truncated

>    { 
>       "_id" : "10091713", ... , "name" : "Surry Hills Studio", ... , "price": 181.00, ...
>    }
>    { 
>       "_id" : "9908871", ... , "name" : "Family friendly beach house", ... , "price": 751.00, ...
>    }
>    { 
>       "_id" : "20989061", ... , "name" : "Big and sunny Narraben room", ... , "price": 60.00, ...
>    }

Reviewer 2: ```suggestion
   ...
   { "_id" : "10091713", ... , "name" : "Surry Hills Studio", ... , "price": 181.00, ... }, 
   { "_id" : "9908871", ... , "name" : "Family friendly beach house", ... , "price": 751.00, ...  }, 
   {  "_id" : "20989061", ... , "name" : "Big and sunny Narraben room", ... , "price": 60.00, ... }, 
   ...
```
Depending on if there are documents before or after, there should be `...` to show theres more that you're not including.

Author: accepted!
